src/modules/cache-client/cache-client.service.ts

```diff
@@ -681,18 +665,48 @@ export class CacheClient implements ICacheClient {
     );
   }
 
-  private async refreshToken(): Promise<AuthTokens | undefined> {
-    if (!this.refresh_token) return undefined;
+  private async refreshToken(): Promise<void> {
+    if (!this.isBrowser && !this.refresh_token) return undefined;
     getLogger().info('[CACHE CLIENT] refreshing tokens');
-    const { data } = await this._httpClient.get<{
-      token: string;
-      refreshToken: string;
-    }>(
+    const res = await this._httpClient.get<AuthTokens>(
       `/refresh_token${
         this.isBrowser ? '' : `?refresh_token=${this.refresh_token}`
       }`
     );
     getLogger().debug('[CACHE CLIENT] refreshed tokens fetched');
-    return data;
+    if (!this.isBrowser) {
+      this.setTokens(res);
+    }
+  }
+
+  /**
+   * Saves access and refresh tokens
+   *
+   * @param res Response from login request
+   */
+  private setTokens({ headers, data }: AxiosResponse) {
+    let token: AuthTokens['token'] | undefined;
+    let refreshToken: AuthTokens['refreshToken'] | undefined;
+    if (headers['set-cookie']) {
+      const cookies = setCookie.parse(headers['set-cookie'], {
+        decodeValues: false,
+        map: true,
+      });
+      const tokenCookie = cookies['token'];
+      const refreshTokenCookie = cookies['refreshToken'];
+
+      if (tokenCookie && refreshTokenCookie) {
+        token = tokenCookie.value;
+        refreshToken = refreshTokenCookie.value;
+      }
+    }
+    if (!token || !refreshToken) {
+      token = data.token;
+      refreshToken = data.refreshToken;
+    }
+    this.refresh_token = refreshToken;
+    this._httpClient.defaults.headers.common[
+      'Authorization'
```

Comment:
so the server is going to expect a token in the Authorization header, not the actual cookie? What about the browser then?

Reply:
Server expects token based on client. When iam lib runs in Node.js this method sets token in header, when iam runs in browser this method isn't called and cookies managed by browser

Reply:
I see. So, the server needs to expect token coming from both `Authorization` and `Cookie` headers. What if both are sent and worse if they are different because of some bug?

Reply:
OK. So, the `Authorization` header takes precedence over Cookie now. 👍